main.tf

```
@@ -351,3 +351,54 @@ resource "aws_security_group_rule" "egress" {
   prefix_list_ids          = lookup(each.value, "prefix_list_ids", null)
   source_security_group_id = lookup(each.value, "source_security_group_id", null)
 }
+
+################################################################################
+# Parameter Group
+################################################################################
+
+locals {
+  cluster_parameter_group_name = try(coalesce(var.db_cluster_parameter_group_name, var.name), null)
+  db_parameter_group_name      = try(coalesce(var.db_parameter_group_name, var.name), null)
+}
+
+resource "aws_rds_cluster_parameter_group" "this" {
+  count = var.create_cluster && var.create_db_cluster_parameter_group ? 1 : 0
+
+  name        = var.db_cluster_parameter_group_use_name_prefix ? null : local.cluster_parameter_group_name
+  name_prefix = var.db_cluster_parameter_group_use_name_prefix ? "${local.cluster_parameter_group_name}-" : null
+  description = var.db_cluster_parameter_group_description
+  family      = var.db_cluster_parameter_group_family
+
+  dynamic "parameter" {
+    for_each = var.db_cluster_parameter_group_parameters
+
+    content {
+      name         = parameter.value.name
+      value        = parameter.value.value
+      apply_method = try(parameter.value.apply_method)
```

```suggestion
      apply_method = try(parameter.value.apply_method, "immediate")
```

Ref - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_cluster_parameter_group#apply_method